xds/src/main/java/io/grpc/xds/XdsNameResolver.java

```diff
@@ -1071,21 +852,30 @@ public void run() {
    */
   private static class RoutingConfig {
     private final long fallbackTimeoutNano;
-    private final List<Route> routes;
-    private final boolean applyFaultInjection;
-    @Nullable
-    private final HttpFault faultConfig;
+    final List<Route> routes;
+    // Null if HttpFilter is not supported.
+    @Nullable final List<NamedFilter> filterChain;
+    final Map<String, FilterConfig> virtualHostOverrideConfig;
 
-    private static final RoutingConfig empty =
-        new RoutingConfig(0L, Collections.<Route>emptyList(), false, null);
+    private static RoutingConfig empty = new RoutingConfig(
+        0L, Collections.<Route>emptyList(), null, Collections.<String, FilterConfig>emptyMap());
 
     private RoutingConfig(
-        long fallbackTimeoutNano, List<Route> routes, boolean applyFaultInjection,
-        HttpFault faultConfig) {
+        long fallbackTimeoutNano, List<Route> routes, @Nullable List<NamedFilter> filterChain,
+        Map<String, FilterConfig> virtualHostOverrideConfig) {
       this.fallbackTimeoutNano = fallbackTimeoutNano;
       this.routes = routes;
-      this.applyFaultInjection = applyFaultInjection;
-      this.faultConfig = faultConfig;
+      if (filterChain == null) {
```

Comment:
Redundant and verbose code, this
```
if (filterChain == null) {
  this.filterChain = null;
}
```
is ugly.

Isn't this just

```
checkArgument(filterChain == null || !filterChain.isEmpty());
this.filterChain = filterChain;
```
Or you can make your implementation more robust by being able to handle an empty list (the non-empty filter chain requirement is unnecessarily strict, and fragile).

The unmodifiable wrapper is optional, as you are not leaking the list to any other classes, you are the only person that uses it within the scope of this class.

Reply:
Simplified.